packages/server-info/src/user/current.ts

```diff
@@ -97,13 +97,19 @@ export const patchCurrent = (
                 return;
             }
 
-            const updatedUser = await database.user.setNickname(user.id, trimmedNickname);
+            const updatedUser = await database.user.setProfileInfo(user.id, trimmedNickname, picture);
+            outputUser = convertDatabaseUserToUserModel(updatedUser);
+        }
+        if (nickname === null) {
```

Comment:
functionality 🔨: there needs to be some checks here to make sure that the picture ID supplied is a valid picture ID